bin/commands/init/index.ts

```diff
@@ -0,0 +1,121 @@
+  common.printLevel0Message(`Configure Zowe`);
+
+  // Validation
+  common.requireZoweYaml();
+
+  // Read job name and validate
+  const zoweConfig = config.getZoweConfig();
+
+  
+  common.printLevel1Message(`Check if need to update runtime directory, Java and/or node.js settings in Zowe YAML configuration`);
+  // node.home
+  let newNodeHome;
+  const configNodeHome=zoweConfig.node?.home;
+  // only try to update if it's not defined
+  if (!configNodeHome || configNodeHome == 'DETECT') {
+    node.requireNode();
+    newNodeHome=std.getenv('NODE_HOME');
+  }
+
+  // java.home
+  let newJavaHome;
+  const configJavaHome=zoweConfig.java?.home;
+  // only try to update if it's not defined
+  if (!configJavaHome || configJavaHome == 'DETECT') {
+    java.requireJava();
+    newJavaHome=std.getenv('JAVA_HOME');
+  }
+
+  // zowe.runtimeDirectory
+  let newZoweRuntimeDir;
+  // do we have zowe.runtimeDirectory defined in zowe.yaml?
+  const configRuntimeDir = zoweConfig.zowe?.runtimeDirectory;
+  if (configRuntimeDir) {
+    if (configRuntimeDir != std.getenv('ZWE_zowe_runtimeDirectory')) {
+      common.printErrorAndExit(`Error ZWEL0105E: The Zowe YAML config file is associated to Zowe runtime "${configRuntimeDir}", which is not same as where zwe command is located.`, undefined, 105);
+    }
+  } else {
+    newZoweRuntimeDir = std.getenv('ZWE_zowe_runtimeDirectory');
+  }
+
+  if (newNodeHome || newJavaHome || newZoweRuntimeDir) {
+    if (std.getenv("ZWE_CLI_PARAMETER_UPDATE_CONFIG") == "true") {
+      let updateObj:any = {};
+      if (newNodeHome) {
+        updateObj.node = {home: newNodeHome};
+      }
+      if (newJavaHome) {
+        updateObj.java = {home: newJavaHome};
+      }
+      if (newZoweRuntimeDir) {
+        updateObj.zowe = {runtimeDirectory: newZoweRuntimeDir};
+      }
+      json.updateZoweYamlFromObj(std.getenv('ZOWE_CLI_PARAMETER_CONFIG'), updateObj);
+
+      common.printLevel2Message(`Runtime directory, Java and/or node.js settings are updated successfully.`);
+    } else {
+      common.printMessage(`These configurations need to be added to your YAML configuration file:`);
+      common.printMessage(``);
+      if (newZoweRuntimeDir) {
+        common.printMessage(`zowe:`);
+        common.printMessage(`  runtimeDirectory: "${newZoweRuntimeDir}"`);
+      }
+      if (newNodeHome) {
+        common.printMessage(`node:`);
+        common.printMessage(`  home: "${newNodeHome}"`);
+      }
+      if (newJavaHome) {
+        common.printMessage(`java:`);
+        common.printMessage(`  home: "${newJavaHome}"`);
+      }
+
+      common.printLevel2Message(`Please manually update "${std.getenv('ZWE_CLI_PARAMETER_CONFIG')}" before you start Zowe.`);
+    }
+  } else {
+    common.printLevel2Message(`No need to update runtime directory, Java and node.js settings.`);
+  }
+
+  initGenerate.execute(dryRun);
```

Comment:
## Help
Missing in `.help`:
```
    This command will run these sub-commands in sequence:

    - `zwe init mvs`
    - `zwe init vsam`
    - `zwe init apfauth`
    - `zwe init security`
    - `zwe init certificate`
    - `zwe init stc`
```

## Dry-run
How do I run `--dry-run` for `zwe init` if `zwe init --config "PARMLIB(YAMLEZWE(ZWEYAML))" --dry-run` is returning `Error ZWEL0102E: Invalid parameter --dry-run`?

Reply:
Done in https://github.com/zowe/zowe-install-packaging/pull/3718/commits/07117559724a383b8155bf279a55a61fecddf0d0